### PR TITLE
Simplify thesaurus functions

### PR DESCRIPTION
**Pull request by Rekyt**
Related to #9 to simplify the names of thesaurus functions.

### CODE REVIEW

File: R/rt_external_db.R [hunk 1]
```diff
@@ -1,29 +1,22 @@
...
 #'
 #' @export
-rt_external_db_id = function(db_id) {
-
-  check_required_arg(db_id, paste0("retrieve a specific external database ",
-                                   "using its id"))
+rt_external_db = function(db_id = NULL) {
 
-  api_query = rt_GET("externalDb/", tolower(db_id))
+  if (is.null(db_id)) {
```
Comment (Bisaloo) [inline R/rt_external_db.R]: No need for the `if ... else` construction here.

`rt_GET("externalDb") == rt_GET("externalDb/", tolower(NULL))`

File: R/rt_operational_groups.R [hunk 2]
```diff
@@ -1,32 +1,19 @@
-#' Retrieve the list of operational groups used in TaxRef
+#' Retrieve operational groups used in TaxRef
 #'
 #' Operational Groups are names of groups either taxonomic or vernacular
 #' used across TaxRef in French
 #'
```
Comment (Bisaloo) [inline R/rt_operational_groups.R]: `#' If used without argument returns the list of all operational groups used in TaxRef.`

File: R/rt_source_uses.R [hunk 3]
```diff
@@ -2,31 +2,23 @@
...
-#'
-#' @param usage_id {`numeric(1)`}\cr{}
-#'                 the id of a source use in TaxRef see
-#'                 [`rt_source_uses()`]
+#' @param usage_id {`numeric(1)`, default = `NULL`}\cr{}
+#'                 the id of a source use in TaxRef
 #'
 #' @examples
-#' rt_source_uses_id(usage_id = 11)
+#' rt_source_uses(276)
```
Comment (Bisaloo) [inline R/rt_source_uses.R]: I think it's better to use the arg name in examples:

```suggestion
#' rt_source_uses(usage_id = 276)
```

File: R/rt_interactions_type.R [hunk 4]
```diff
@@ -1,35 +1,21 @@
...
-#'                  A character identifying the type of interactions (see
-#'                  [`rt_interactions_type()`] for a list of interactions types)
+#' @param interaction_id {`character(1)`, default = `NULL`}\cr{}
+#'                  A character identifying the type of interactions
 #'
 #' @examples
-#' rt_interactions_type_id("PREDATEUR")
-#'
-#' @seealso see [`rt_interactions_type()`] for a list of interactions types
+#' rt_interactions_type("PREDATEUR")
```
Comment (Bisaloo) [inline R/rt_interactions_type.R]:
```suggestion
#' rt_interactions_type(interaction_id = "PREDATEUR")
```

Comment (Bisaloo) [reply R/rt_interactions_type.R]: Does the case matter here? Other functions use `tolower()`

File: R/rt_taxo_ranks.R [hunk 5]
```diff
@@ -1,28 +1,22 @@
...
 #'
-#' @param taxo_id {`character(1)`}\cr{}
-#'                the id of a taxonomic rank in TaxRef see
-#'                [`rt_taxo_ranks()`]
+#' @param taxo_id {`character(1)`, default = `NULL`}\cr{}
+#'                the id of a taxonomic rank in TaxRef
 #'
 #' @examples
-#' rt_taxo_ranks_id(taxo_id = "VAR")
+#' rt_taxo_ranks(taxo_id = "VAR")
```
Comment (Bisaloo) [inline R/rt_taxo_ranks.R]: Same comment about the case and  `tolower()`

File: R/rt_interactions_type.R (hunk 4 above)
Comment (Rekyt) [reply R/rt_interactions_type.R]: For all TaxRef functions, case does matter. However as everything is uppercase here, we can use `toupper()` to make it case-insensitive.

File: R/rt_external_db.R (hunk 1 above)
Comment (Rekyt) [reply R/rt_external_db.R]: Much simpler indeed, will change for all functions

File: R/rt_source_uses.R (hunk 3 above)
Comment (Rekyt) [reply R/rt_source_uses.R]: Yes I've missed this one.

File: R/rt_taxo_ranks.R (hunk 5 above)
Comment (Rekyt) [reply R/rt_taxo_ranks.R]: In this case the arguments can be mixed case so it is impossible to use `tolower()` nor `toupper()` (the only exception being the `Dumm`  case)

File: R/rt_taxo_ranks.R [hunk 6]
```diff
@@ -1,26 +1,18 @@
...
 #'
 #' @export
-rt_taxo_ranks_id = function(taxo_id) {
+rt_taxo_ranks = function(taxo_id = NULL) {
 
-  check_required_arg(taxo_id, "retrieve a specific taxonomic rank using its id")
+  taxo_id = gsub("DUMM", "Dumm", toupper(taxo_id))
```
Comment (Bisaloo) [inline R/rt_taxo_ranks.R]: Maybe add `^` and `+` for more security? (to prevent partial matching in other keywords)

**Pull request merged**